tests/phpunit/test-page-cache.php

```diff
@@ -0,0 +1,195 @@
+		$input = [
+			'default_ttl' => 100, 
+			'maintenance_mode' => 'disabled',
+		];
+		$output = $this->pantheon_cache->sanitize_options( $input );
+		$this->assertEquals( 100, $output['default_ttl'] );
+
+		// Test case where default_ttl is below 60 and in live environment.
+		$_ENV['PANTHEON_ENVIRONMENT'] = 'live';
+		$input = [ 
+			'default_ttl' => 30, 
+			'maintenance_mode' => 'disabled',
+		];
+		$output = $this->pantheon_cache->sanitize_options( $input );
+		$this->assertEquals( 60, $output['default_ttl'] );
+	}
+
+	/**
+	 * Test sanitize_options() with valid and invalid maintenance_mode values.
+	 */
+	public function test_sanitize_maintenance_mode() {
+		// Test with valid maintenance_mode values.
+		foreach ( [ 'anonymous', 'everyone' ] as $mode ) {
+			$input = [
+				'default_ttl' => 600, 
+				'maintenance_mode' => $mode,
+			];
+			$output = $this->pantheon_cache->sanitize_options( $input );
+			$this->assertEquals( $mode, $output['maintenance_mode'] );
+		}
+
+		// Test with invalid maintenance_mode value.
+		$input = [
+			'default_ttl' => 600, 
+			'maintenance_mode' => 'invalid_value',
+		];
+		$output = $this->pantheon_cache->sanitize_options( $input );
+		$this->assertEquals( 'disabled', $output['maintenance_mode'] );
+	}
+
+	/**
+	 * Test sanitize_options() with empty or missing values.
+	 */
+	public function test_sanitize_empty_or_missing_values() {
+		$_ENV['PANTHEON_ENVIRONMENT'] = 'live';
+
+		// Test with missing keys.
+		$input = [
+			'default_ttl' => '',
+			'maintenance_mode' => null,
+		];
+
+		$expected_output = [
+			'default_ttl' => 60, // Default TTL is set to 60 on live environments.
+			'maintenance_mode' => 'disabled',
+		];
+		$output = $this->pantheon_cache->sanitize_options( $input );
+		$this->assertEquals( $expected_output, $output );
+
+		// Test with empty maintenance_mode.
+		$input = [ 
+			'default_ttl' => 600, 
+			'maintenance_mode' => '', 
+		];
+		$output = $this->pantheon_cache->sanitize_options( $input );
+		$this->assertEquals( 'disabled', $output['maintenance_mode'] );
+	}
+	
+	/**
+	 * Test flush_site() with an unauthorized user.
+	 */
+	public function test_flush_site_unauthorized_user() {
+		// Mock current_user_can to return false.
+		wp_set_current_user( 0 ); // No permissions.
+
+		$result = $this->pantheon_cache->flush_site();
+		$this->assertFalse( $result );
+	}
+
+	/**
+	 * Test flush_site() with an authorized user but an invalid nonce.
+	 */
+	public function test_flush_site_authorized_user_invalid_nonce() {
+		// Mock current_user_can to return true.
+		$user_id = $this->factory->user->create( [ 'role' => 'administrator' ] );
+		wp_set_current_user( $user_id );
+
+		$_POST['pantheon-cache-nonce'] = 'invalid_nonce';
+
+		$result = $this->pantheon_cache->flush_site();
+		$this->assertNull( $result ); // No action should be taken.
+	}
+
+	/**
+	 * Test enqueue_urls() with valid and invalid URLs.
+	 */
+	public function test_enqueue_urls() {
```

Comment:
seems like this should be a table driven test. optional.

Reply:
table driven tests would be great but already sunk more than 2 points into these 😅